src/core/ipfs/files.js

```diff
@@ -81,6 +82,10 @@ module.exports = function files (self) {
           return file
```

Comment:
It's weird to me that it starts flowing before a read on a 'data' event is attached.